### PR TITLE
feat: switch to tsd

### PR DESCRIPTION
**Pull request by zekth**
fix: #21

### CODE REVIEW

**Comment by SimenB** [comment]
Why are most of the tests deleted instead of migrated?

**Comment by zekth** [comment]
> Why are most of the tests deleted instead of migrated?

added missing assertions. But most of the test files were compilations tests aren't they? Do i miss something in here?

**Comment by SimenB** [comment]
Yeah, seems to have been quite a bit of duplication

File: types/test/basic.ts [hunk 1]
```diff
@@ -1,58 +0,0 @@
...
-import cookie from 'fastify-cookie';
-import session from '../../';
-
-const app = fastify();
-
-declare module 'fastify' {
-  interface Session {
-    id?: number;
-  }
-}
```
Comment (SimenB) [inline types/test/basic.ts:11]: this part is missing - adding own properties to the session object

File: types/test/basic.ts [hunk 2]
```diff
@@ -1,58 +0,0 @@
...
-  async handler(_req, _rep) {
-    return { status: 'ok' };
-  },
-});
-
-app.route({
-  method: 'GET',
-  url: '/logout',
-  preHandler(req, _rep, next) {
-    req.destroySession(next);
```
Comment (SimenB) [inline types/test/basic.ts:36]: this function is not asserted on

File: types/test/session-type.ts [hunk 3]
```diff
@@ -1,65 +0,0 @@
...
-import session from '../../';
-
-declare module 'fastify' {
-  interface Session {
-    user?: {
-      id: number;
-    };
-  }
-}
-
```
Comment (SimenB) [inline types/test/session-type.ts:12]: same here I guess, but a deeper object. Both should get tests I guess?

File: types/types.d.ts [hunk 4]
```diff
@@ -92,10 +92,7 @@ declare namespace FastifySessionPlugin {
 export class MemoryStore implements FastifySessionPlugin.SessionStore {
   constructor(map?: Map<string, any>);
   set(sessionId: string, session: any, callback: (err?: Error) => void): void;
```
Comment (SimenB) [inline types/types.d.ts:94]: why is `session` typed as `any` instead of `Session`?

**Review by SimenB — CHANGES_REQUESTED** [review]
We need to assert the interface merging of the session object works, beyond that LGTM 👍

File: types/types.d.ts (hunk 4 above)
Comment (zekth) [reply types/types.d.ts:94]: This is inherited from previous code. We can fix it in another PR.

**Review by mcollina — APPROVED** [review]
lgtm

File: types/types.test-d.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  destroy(_sessionId: string, _callback: Function) {}
+}
+
+declare module "fastify" {
+  interface Session {
+    sessionId: string;
+    encryptedSessionId: string;
+    touch(): void;
+    regenerate(): void;
```
Comment (SimenB) [inline types/types.test-d.ts:23]:
```suggestion
```

should merge

**Review by SimenB — APPROVED** [review]

**Pull request merged**